R/add_lcsi.R

```diff
@@ -1,20 +1,179 @@
+#' not_applicable_val = "Not Applicable")
+#'
+#' }
+add_lcsi <- function(.dataset,
+                     lcsi_stress_vars,
+                     lcsi_crisis_vars,
+                     lcsi_emergency_vars,
+                     yes_val = NULL,
+                     no_val = NULL,
+                     exhausted_val = NULL,
+                     not_applicable_val = NULL
+) {
+
+df <- .dataset
+
+  #If NULL, set a standard default values for LCSI responses.
+
+  if(is.null(yes_val)) {yes_val = "1"}
+  if(is.null(yes_val)) {no_val = "2"}
+  if(is.null(yes_val)) {exhausted_val = "3"}
+  if(is.null(yes_val)) {not_applicable_val = "4"}
+
+  # Check 1: correct # of stress vars
+  if(length(lcsi_stress_vars) != 4) {
+    stop(paste0("Need 4 seperate 'stress' livelihood coping strategies. There were ", length(lcsi_stress_vars), " variables were given for 'stress'." ))
+  }
+
+  # Check 2: correct # of crisis vars
+  if(length(lcsi_crisis_vars) != 3) {
+    stop(paste0("Need 3 seperate 'crisis' livelihood coping strategies. There were ", length(lcsi_crisis_vars), " variables were given for 'crisis'." ))
+  }
+
+  # Check 3: correct # of emergency vars
+  if(length(lcsi_emergency_vars) != 3) {
+    stop(paste0("Need 3 seperate 'emergency' livelihood coping strategies. There were ", length(lcsi_emergency_vars), " variables were given for 'emergency'." ))
+  }
+
+  lcs_codes <- df %>% dplyr::select(c(lcsi_stress_vars, lcsi_crisis_vars, lcsi_emergency_vars)) %>% t %>% c %>% unique
+
+  # Check 4: Number of unique values across LCSI variables must be <= 4.
+
+  if(length(lcs_codes) > 4) {
+    stop(paste0("There are ", length(lcs_codes), " in your LCSI variables, but there should only be 4 standard values (Yes, No, Exhausted, Not Applicable). Please check your inputs.s"))
+  }
+
+  # Check 5: There are 4 unique response values, but one of them is not counted in your response inputs.
+
+  a <- yes_val %in% lcs_codes
+  b <- no_val %in% lcs_codes
+  c <- exhausted_val %in% lcs_codes
+  d <- not_applicable_val %in% lcs_codes
+
+  e <- any(lcs_codes %in% yes_val)
+  f <- any(lcs_codes %in% no_val)
+  g <- any(lcs_codes %in% exhausted_val)
+  h <- any(lcs_codes %in% not_applicable_val)
+
+  if(length(lcs_codes) == 4 & !all(c(a,b,c,d,e,f,g,h))) {
+    print(paste0("There are 4 unique response values in your input, but at least one of them does not match your expected values. Please check your input."))
+    print(paste0("Values in your dataset: "))
+    print(paste0(lcs_codes))
+    print(cat(paste0("Values you expected, or default values: \n",
+                     "Yes: ", yes_val, " \n",
+                     "No: ", no_val, " \n",
+                     "Exhausted: ", exhausted_val, " \n",
+                     "Not applicable: ", not_applicable_val, "\n")))
+    stop("Please check your expected, or default, values match what is in your dataset.")
+  }
+
+  # Warning 1: If < 4 unique responses observed, and
+
+  if(length(lcs_codes) < 4 & !all(a,b,c,d)) {
```

Comment:
Can you also check that the lcs codes are different? 
For example I was able to call the function with these params and it went through 
`add_lcsi(.dataset = input_data1,
         lcsi_stress_vars = c("stress1", "stress2", "stress3", "stress4"),
         lcsi_crisis_vars = c("crisis1", "crisis2", "crisis3"),
         lcsi_emergency_vars = c("emergency1", "emergency2", "emergency3"),
         yes_val = "Yes",
         no_val = "No",
         exhausted_val = "Exhausted",
         not_applicable_val = "Exhausted")`